### PR TITLE
feat: Streamline transformations and minor component improvements

### PR DESCRIPTION
**Pull request by danielptv**
[ODDSK-200](https://h-ka-team-rdqzrlfpomci.atlassian.net/browse/ODDSK-200).

We decided to not internationalize numbers as there are multiple edge cases that cannot be handled without significant configuration effort for every visualization.

### CODE REVIEW

File: src/transformations/transformType.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    if (!Number.isNaN(parsedValue)) {
+                        return [key, parsedValue];
+                    }
+                    return [key, value];
+                }),
+            ) as Record<string, string | number | boolean>,
+    );
+}
+
+function parseGermanNumberToInternationalFormat(value: string) {
```
Comment (Zerdados) [inline src/transformations/transformType.ts:28]: We can probably simplify this function:
`const parsedValue = Number(value.replace('.', '').replace(',', '.'));`
`return !Number.isNaN(parsedValue) ? parsedValue : Number.NaN;`

File: src/messages/en.json [hunk 2]
```diff
@@ -1,58 +1,58 @@
...
-  "Legend": {
-    "title": "Legend"
-  },
-  "ResetView": {
-    "resetTooltip": "Reset view"
-  },
-  "AxisSelector": {
-    "title": "Select X-Axis:"
-  }
+    "AxisSelector": {
```
[the remark this answers is not in the document]
Comment (danielptv) [reply src/messages/en.json]: Yes, in the future we may want to do it like this: https://next-intl-docs.vercel.app/docs/usage/messages#messages-nested-structure

File: src/components/visualization/chart-table-filter/ClearableInputGroup.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { ChangeEvent, ReactNode } from 'react';
-import { useTranslations } from 'next-intl';
+import { useLocale, useTranslations } from 'next-intl';
```
[the remark this answers is not in the document]
Comment (danielptv) [reply src/components/visualization/chart-table-filter/ClearableInputGroup.tsx:2]: Not really, the `input` was using it but it wasn't making any difference. Removed.

File: src/transformations/transformType.ts (hunk 1 above)
Comment (danielptv) [reply src/transformations/transformType.ts:28]: Good point

**Review by Zerdados — APPROVED** [review]

**Pull request merged**